src/Processor/Braintree/Mapper/Subscription/ModifierMapper.php

```diff
@@ -102,6 +102,43 @@ private function withRemovedPreviousPlanDefaults(array &$request, array $default
         return $request;
     }
 
+    /**
+     * Determines if a modifier should be added or modified on an existing Subscription based off of the Plan defaults & which modifiers are already on the Subscription.
+     */
+    private function isModifierAddedOnExistingSubscription(bool $isDefaultModifier, Subscription $subscription, ModifierType $modifierType, Modifier $modifier): bool
+    {
+        $hasFnName = $modifierType->equals(ModifierType::addOn()) ? 'hasAddOn' : 'hasDiscount';
+        $previousPlanFnName = $modifierType->equals(ModifierType::addOn()) ? 'getPreviousAddOns' : 'getPreviousDiscounts';
+
+        /**
+         * @var bool
+         */
+        $planHasModifier = $subscription->getPlan()->{$hasFnName}($modifier->getId());
+
+        /**
+         * @var Modifier[]
+         */
+        $subscriptionPreviousModifiers = $subscription->{$previousPlanFnName}();
+
+        $isAdding = $isDefaultModifier || $planHasModifier;
+
+        if (!$isAdding) {
+            return false;
+        }
+
+        $isModifierOnExistingSubscription = array_reduce(
```

Comment:
Can we just return `! array_reduce`?

Reply:
I don't see why we can't. Makes sense to me.